routers/api/v1/repo/issue_comment.go

```diff
@@ -169,6 +170,11 @@ func CreateIssueComment(ctx *context.APIContext, form api.CreateIssueCommentOpti
 		return
 	}
 
+	if issue.IsLocked && !ctx.Repo.CanWrite(models.UnitTypeIssues) && !ctx.User.IsAdmin {
+		ctx.Error(500, "CreateIssueComment", errors.New(ctx.Tr("repo.issues.comment_on_locked")))
```

Comment:
This shouldn't be a 500 if the user does not have the right to do it. Forbidden would be better (403)